datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/mappers/DatasetUpdateInputMapper.java

```diff
@@ -36,9 +44,53 @@ public Dataset apply(@Nonnull final DatasetUpdateInput datasetUpdateInput) {
         }
 
         if (datasetUpdateInput.getInstitutionalMemory() != null) {
-            result.setInstitutionalMemory(InstitutionalMemoryUpdateMapper.map(datasetUpdateInput.getInstitutionalMemory()));
+            result.setInstitutionalMemory(
+                    InstitutionalMemoryUpdateMapper.map(datasetUpdateInput.getInstitutionalMemory()));
+        }
+
+        if (datasetUpdateInput.getGlobalTags() != null) {
+            final GlobalTags globalTags = new GlobalTags();
+            globalTags.setTags(
+                    new TagAssociationArray(
+                            datasetUpdateInput.getGlobalTags().getTags().stream().map(
+                                    element -> TagAssociationUpdateMapper.map(element)
+                            ).collect(Collectors.toList())
+                    )
+            );
+            result.setGlobalTags(globalTags);
+        }
+
+        if (datasetUpdateInput.getEditableSchemaMetadata() != null) {
+            final EditableSchemaMetadata editableSchemaMetadata = new EditableSchemaMetadata();
+            editableSchemaMetadata.setEditableSchemaFieldInfo(
+                    new EditableSchemaFieldInfoArray(
+                            datasetUpdateInput.getEditableSchemaMetadata().getEditableSchemaFieldInfo().stream().map(
+                                    element -> mapSchemaFieldInfo(element)
+                            ).collect(Collectors.toList())));
+            result.setEditableSchemaMetadata(editableSchemaMetadata);
+
         }
 
         return result;
     }
+
+    private EditableSchemaFieldInfo mapSchemaFieldInfo(
+            final com.linkedin.datahub.graphql.generated.EditableSchemaFieldInfoUpdate schemaFieldInfo
+    ) {
+        final EditableSchemaFieldInfo output = new EditableSchemaFieldInfo();
+
+        if (schemaFieldInfo.getDescription() != null) {
```

Comment:
Is this null check necessary?

Reply:
surprisingly yes- i got null pointer exceptions without it